classes/s3_file_system.php

```diff
@@ -40,4 +43,26 @@ protected function get_external_client($config) {
         $s3client = new s3_client($config);
         return $s3client;
     }
+
+    public function copy_object_from_local_to_external_path($contenthash) {
+
+        $config = get_objectfs_config();
+        $s3client = $this->get_external_client($config);
+
+        $localpath = $this->get_local_path_from_hash($contenthash);
+        $externalpath = $s3client->get_filepath_from_hash($contenthash);
+
+        $uploader = new MultipartUploader($s3client->get_client(), $localpath, [
```

Comment:
All direct s3 logic should inside s3_client  itself. Would wrap this in a s3_client method such that code looks like:

$s3client->copy_object_from_local_to_external_path($localpath, $contenthash); 

get_filepath_from_hash can also remain private this way.